extras/create_flir_videos.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import os
+from typing import Dict
+
+import cv2
+import numpy as np
+import yaml
+
+from neurobooth_os.log_manager import make_db_logger
+from neurobooth_os import config
+
+
+def _read_bytes_to_avi(images_filename: str, video_out: cv2.VideoWriter, height, width, depth) -> None:
+    """
+    Reads the file containing the raw images and produces an AVI file encoded as MJPEG
+    Parameters
+    ----------
+    images_filename  Name of file used to store the row frame data
+    video_out        CV2 video writer
+    height           frame height in pixels
+    width            frame depth in pixels
+    depth            frame depth
+
+    Returns
+    -------
+    None
+    """
+    with open(images_filename, "rb") as f:
+        byte_size = height * width * depth
+        while True:
+            chunk = f.read(byte_size)
+            if chunk:
+                bytes_1d = np.frombuffer(chunk, dtype=np.uint8)
+                frame = np.reshape(bytes_1d, newshape=(height, width, depth))
+                video_out.write(frame)
+            else:
+                return
+
+
+def run_conversion() -> None:
+    """
+    Runs raw image file to AVI conversion for all image files in the local_data_dir specified in config file
+
+    Returns
+    -------
+    None
+    """
+
+    config.load_config()
+    logger = make_db_logger()
+    data_folder = config.neurobooth_config.acquisition.local_data_dir
+    subfolders = [f.path for f in os.scandir(data_folder) if f.is_dir()]
+
+    try:
+        for folder in subfolders:
+            logger.info(f'FLIR: Starting conversion in {folder}')
+            manifests = []
+            for file in os.listdir(folder):
+                if file.endswith("flir_manifest.yaml"):
+                    manifests.append(os.path.join(folder, file))
+
+            for manifest_filename in manifests:
+                with open(manifest_filename, 'r') as file:
+                    manifest: Dict = yaml.safe_load(file)
+                    image_filename = manifest["image_file"]
```

Comment:
The if-else block does not need to be inside the with file handler block, correct? If so, I'd indent it out and keep if-else block separate from reading in yaml. That way if there's an error and the try doesn't work, at least the yaml stays unaffected.

Reply:
Unless I'm missing something, the if else block prevents the manifest from being deleted unless the video file has been written. I think it's fine the way it is.